src/status_im/contexts/wallet/events.cljs

```diff
@@ -538,3 +561,27 @@
             :on-error   #(log/info "failed to fetch crypto on ramps"
                                    {:error %
                                     :event :wallet/get-crypto-on-ramps})}]]]}))
+
+(rf/reg-event-fx
+ :wallet/process-keypair-from-backup
+ (fn [{:keys [db]} [{:keys [backedUpKeypair]}]]
+   (let [{:keys [key-uid accounts]} backedUpKeypair
+         keypairs-db                (get-in db [:wallet :keypairs])
+         updated-keypairs           (-> (filter #(not= (:key-uid %) key-uid) keypairs-db)
+                                        (conj backedUpKeypair)
+                                        data-store/rpc->keypairs)
+         accounts-fx                (mapv (fn [{:keys [chat] :as account}]
```

Comment:
would it not be easier to understand to do a filter first and then fire the dispatch for the set of accounts that meet the criteria? 🤔

Reply:
I totally agree. There will be only one chat account in the whole key pair data. Filtering and then looping over that result is a bit time-consuming considering we can do the filter at once in `fx` as it accepts `nil`.